crates/scroll/alloy/evm/src/block/mod.rs

```diff
@@ -129,6 +131,22 @@ where
             };
         }
 
+        // apply gas oracle predeploy upgrade at Feynman transition block.
+        if self
+            .spec
+            .scroll_fork_activation(ScrollHardfork::Feynman)
+            .active_at_timestamp(self.evm.block().timestamp)
```

Comment:
could we use `transitions_at_timestamp(self.evm.block().timestamp, self.evm.block().timestamp.saturated_sub(1))`, which would mean we only enter the branch for a timestamp equal to the Feynman block, but never after, avoiding having to pass the witness for every Feynman block?

Reply:
> I think we can limit the read on `IS_FEYNMAN_SLOT` to the block timestamps where `ts == FEYNMAN_TS_TRANSITION`. What do you think?

The timestamp is not guaranteed to match the fork timestamp, we only know that it will be `>=`, but in extreme cases it could be much larger, e.g. if the sequencer is down.

Furthermore, there can be multiple consecutive blocks with the same timestamp.

Reply:
> could we use `transitions_at_timestamp(self.evm.block().timestamp, self.evm.block().timestamp.saturated_sub(1))`, which would mean we only enter the branch for a timestamp equal to the Feynman block, but never after, avoiding having to pass the witness for every Feynman block?

Same issue as above. Though we can consider using an exact timestamp for the transition block in l2geth.